src/org/zaproxy/zap/view/StandardFieldsDialog.java

```diff
@@ -1656,18 +1691,122 @@ public void setTabsVisible(String[] tabLabels, boolean visible) {
+			return;
+		}
+
+		if (scrollable) {
+			if (isTabScrollable(tabPanel)) {
+				return;
+			}
+
+			String title = Constant.messages.getString(tabLabel);
+			int tabIndex = tabbedPane.indexOfTab(title);
+			boolean selected = tabbedPane.getSelectedIndex() == tabIndex;
+
+			JScrollPane scrollPane = createTabScrollable(tabPanel);
+			if (scrollPane == null) {
+				return;
+			}
+			panelToScrollPaneMap.put(tabPanel, scrollPane);
+
+			if (tabIndex == -1) {
+				return;
+			}
+
+			tabbedPane.insertTab(title, null, scrollPane, null, tabIndex);
+			if (selected) {
+				tabbedPane.setSelectedIndex(tabIndex);
+			}
+			return;
+		}
+
+		if (!isTabScrollable(tabPanel)) {
+			return;
+		}
+
+		String title = Constant.messages.getString(tabLabel);
+		int tabIndex = tabbedPane.indexOfTab(title);
+		tabbedPane.insertTab(title, null, tabPanel, null, tabIndex);
+		tabbedPane.removeTabAt(tabIndex + 1);
+		panelToScrollPaneMap.remove(tabPanel);
+	}
+
+	/**
+	 * Tells whether or not the given panel is scrollable.
+	 * <p>
+	 * <strong>Note:</strong> The scrollable state returned by this mehtod only applies to tabs that were set to be (or not)
+	 * scrollable through the method {@link #setTabScrollable(String, boolean)}, not to "panels" added directly to a tab with
+	 * {@link #setCustomTabPanel(int, JComponent)}.
+	 *
+	 * @param tabPanel the panel to check.
+	 * @return {@code true} if the tab is scrollable, {@code false} otherwise.
+	 */
+	private boolean isTabScrollable(JPanel tabPanel) {
+		return panelToScrollPaneMap.containsKey(tabPanel);
+	}
+
+	/**
+	 * Tells whether or not the tab with the given label is scrollable.
+	 * <p>
+	 * <strong>Note:</strong> The scrollable state returned by this mehtod only applies to tabs that were set to be (or not)
+	 * scrollable through the method {@link #setTabScrollable(String, boolean)}, not to "panels" added directly to a tab with
+	 * {@link #setCustomTabPanel(int, JComponent)}.
+	 * 
+	 * @param tabLabel the label of the tab to check.
+	 * @return {@code true} if the tab is scrollable, {@code false} otherwise.
+	 * @since TODO add version
+	 */
+	protected boolean isTabScrollable(String tabLabel) {
+		JPanel tabPanel = this.tabNameMap.get(tabLabel);
+		if (tabPanel == null) {
+			return false;
+		}
+		return isTabScrollable(tabPanel);
+	}
+
+	/**
+	 * Creates and returns a {@link JScrollPane} for the given panel. Called when a tab is
+	 * {@link #setTabScrollable(String, boolean) set to be scrollable}.
+	 * <p>
+	 * By default this method returns a {@code JScrollPane} that has the vertical scrollbar shown as needed and the horizontal
+	 * scrollbar never shown.
+	 *
+	 * @param tabPanel the panel of the tab that should be scrollable, never {@code null}.
+	 * @return the JScrollPane
+	 * @since TODO add version
+	 */
+	protected JScrollPane createTabScrollable(JPanel tabPanel) {
+		return new JScrollPane(tabPanel, JScrollPane.VERTICAL_SCROLLBAR_AS_NEEDED, JScrollPane.HORIZONTAL_SCROLLBAR_NEVER);
```

Comment:
Changes done (both scrollbars are shown as needed and the label of the panel is provided when creating the JScrollPane).